### PR TITLE
Blocks: Split frontend block styles into `block.scss` files

### PR DESCRIPTION
**Pull request by youknowriad**
This PR builds on top of #1418 and splits all the blocks' styles into two files 

 - `block.scss` for styles shared with the frontend
 - `style.scss` for styles only loaded in the editor.

This should allow themes to support the gutenberg blocks without any change. The styles can be overriden by themes.

**Testing instructions**

Try the following blocks and ensure they're showing up properly on the frontend and the backend:

 - button
 - cover image
 - pullquote
 - quote
 - separator

### CODE REVIEW

File: blocks/library/quote/block.scss [hunk 1]
```diff
@@ -17,17 +17,18 @@
...
-	border-left: 4px solid $black;
-	padding-left: 1em;
-}
 
-.editor-visual-editor .blocks-quote-style-2 {
-	padding: 0 1em;
-	& > .blocks-editable p {
-		font-size: 24px;
-		font-style: italic;
+	&.blocks-quote-style-1 {
```
Comment (mtias) [inline blocks/library/quote/block.scss]: just a note, we should clean these classes and make them proper modifiers.

File: blocks/library/quote/block.scss [hunk 2]
```diff
@@ -17,17 +17,18 @@
...
+		border-left: 4px solid $black;
+		padding-left: 1em;
+	}
+
+	&.blocks-quote-style-2 {
+		padding: 0 1em;
+		& > .blocks-editable p {
```
Comment (mtias) [inline blocks/library/quote/block.scss]: We should make editable have a `wp-blocks-editable` class

**Comment by mtias** [comment]
More specific styles needed for:

image

We should make the dash in quotes not be absolutely positioned:

image

File: blocks/library/quote/block.scss (hunk 2 above)
Comment (youknowriad) [inline blocks/library/quote/block.scss]: This was an error I should move this back to `style.scss` and use `> p` instead for frontend.

**Comment by mtias** [comment]
We also need to get the `object-fit` styles from galleries that was merged earlier today.

**Comment by youknowriad** [comment]
@mtias the `is-croped` className was missing in the `save` function. Added in 705c835

**Comment by mtias** [comment]
Now the fixtures will hunt you...

**Comment by jasmussen** [comment]
Nice work!

I think we'll want to do a run on theme styles and polish all of these as a separate PR, but I think it would be great to get this in soon, especially since I did some polish work to many of the blocks in https://github.com/WordPress/gutenberg/pull/1597.

**Comment by youknowriad** [comment]
Ok. I'll wait for the tests to pass and merge. :)

**Pull request merged**